Add broken mirrors (venthyr)

Dathwada opened the pull request:
The mirrors will only be displayed if the character has done the quest "Repair and Restore" (ID: 59740).

For now i don't have a solution to figure out which group of mirrors is active today.

Dathwada committed:
Add broken mirrors (venthyr)

Krovikan committed:
Translated feature-broken-mirrors Shadowlands esES

zarillion commented:
I will take a look at this PR this weekend.

Emmaleah commented:
I came by to redo my pull with QID's but, this looks like a better solution since you took the time to add the drops and all.

zarillion committed:
Move mirrors before Misc group

zarillion committed:
Broken mirror tweaks

Some changes to the way we display these mirrors.

  * Ensure the group is disabled for other covenants. They might
    complete the prereq quest, then change to a different covenant.
  * Move the quest+notes into a data structure before initialization.
    This allows us to reference that data structure in the IsCompleted()
    function to check if a different mirror group has completed quests.
  * Change the rlabel to the covenant icon. I think the color + fgroup
    is enough to indicate what the groupings are.
  * Add a note about how the mirrors work.
  * Increase the size of the icons a bit, they are pretty small.
